subdomain-replace.go

```diff
@@ -0,0 +1,115 @@
+package rdns
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+// Replace is a resolver that modifies queries according to regular expressions
+// and forwards the modified queries to another resolver. Responses are then
+// mapped back to the original query string.
+type SubDomainReplace struct {
+	id       string
+	resolver Resolver
+	exp      subDomainReplaceExpressions
+}
+
+var _ Resolver = &SubDomainReplace{}
+
+type subDomainReplaceExp struct {
+	from string
+	to   string
+}
+
+type subDomainReplaceExpressions []subDomainReplaceExp
+func (r subDomainReplaceExp)  substitute (qname string) (string, bool) {
+	 fromDomain := r.from
+	toDomain := r.to
+	// from {}.ctptech.dev to {}.charlesp.tech
+	// from test.ctptech.dev to test.charlesp.tech
+	fromDomain = strings.Replace(fromDomain, "{}", "", 1)
```

Comment:
Since this only works for the last part of a domain anyway, you wouldn't even need the `{}` placeholder at all. It could simply be `from=some.domain.net to=example.com`